lib/s3-s3.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * AWS.S3-like behavior with failover to a secondary S3 location
+ */
+
+'use strict';
+
+// Used in init() below, this function mimics AWS.S3
+var s3 = function (primaryS3, secondaryS3) {
+  var
+    // request object that proxies to the real AWS.Request object
+    Request = function(realRequest) {
+      var
+        // storage of on() callbacks
+        onFunc = {},
+        // proxying on() callbacks to the real AWS.Request object
+        setupOn = function(realRequest, onType, onFunc) {
+          realRequest.on(onType, function(response) {
+            if (onFunc.hasOwnProperty(onType)) {
+              onFunc[onType](response);
+            }
+          });
+        };
+
+      return {
+        // simply store all of the on() callbacks when they come in
+        on: function (type, callback) {
+          onFunc[type] = callback;
+          return this;
+        },
+        // send on the request to the real request handler
+        send: function() {
+          if (this.hasOwnProperty('params')) {
+            realRequest.params = this.params;
+          }
+          // add on() calls on the real request object, and use the stored on() callbacks
+          // wherever possible
+          setupOn(realRequest, 'send', onFunc);
+          setupOn(realRequest, 'retry', onFunc);
+          setupOn(realRequest, 'extractError', onFunc);
+          setupOn(realRequest, 'extractData', onFunc);
+          setupOn(realRequest, 'success', onFunc);
+          setupOn(realRequest, 'error', onFunc);
+          setupOn(realRequest, 'complete', onFunc);
+
+          realRequest.send();
+        }
+      };
+    };
+
+  return {
+    /**
+     * Mimics putObject in AWS.S3
+     */
+    putObject: function() {
```

Comment:
There's two ways to work with these functions.  One is to get a AWS.Request object back and do all the work there.  The other way is to pass in params (a params object and callback).  We do our calls using the first approach, so that's what I'm supporting here.  It would be nice though to throw an error on the unsupported case... simple enough, will do that.  (It's possible I missed a case and we do actually use the second approach, in which case I'll see it faster and do the extra work.)